pkg/tracing/trace_context.go

```diff
@@ -0,0 +1,46 @@
+package tracing
+
+import (
+	"net/http"
+
+	"github.com/opentracing/opentracing-go"
+	"github.com/openzipkin/zipkin-go-opentracing"
+)
+
+type TraceContext interface {
+	Close() error
+	InitRequestSpan(r *http.Request) (FinishableSpan, error)
```

Comment:
I'm not sure I see where these abstractions lead us to, as it actually is http specific given this function signature